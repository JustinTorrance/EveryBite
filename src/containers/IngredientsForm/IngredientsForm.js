import React, { Component } from 'react'
import { connect } from 'react-redux'
import { fetchRecipes } from '../../thunks/fetchRecipes'
import Button from '@material-ui/core/Button'
import Input from '@material-ui/core/Input'
import Typography from '@material-ui/core/Typography'
import { withRouter } from 'react-router-dom'


export class IngredientsForm extends Component {
  constructor() {
    super()
    this.state = {
      ingredient1: '',
      ingredient2: '',
      ingredient3: '',
      ingredient4: '',
      ingredient5: '',
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  handleSubmit = (e) => {
    e.preventDefault()
    const { history } = this.props
    history.push('/recipes')
    const ingredients = Object.values(this.state).filter(ingredient => {
      if (ingredient !== '') {
        return ingredient
      }
    })
    this.props.fetchRecipes(ingredients)
  }

  render() {
    const { ingredient1, ingredient2, ingredient3, ingredient4, ingredient5 } = this.state
    return(
      
      <section className='ingredient-input-section'>
        <Typography variant="title" color="inherit">
          <div className='instructions'>
            <p>It's easy to begin!</p>
            <p>Simply enter the ingredients you have on hand</p>
            <p>We'll find recipes that include all of your ingredients.</p>
          </div>
        </Typography>
        <form className='ingredients-form' onSubmit={(e) => this.handleSubmit(e)}>
          <Input 
            value={ingredient1}
            name='ingredient1'
            onChange={this.handleChange}
            placeholder='enter an ingredient'
          />
          <Input 
            value={ingredient2}
            name='ingredient2'
            onChange={this.handleChange}
            placeholder='enter an ingredient'
          />
          <Input 
            value={ingredient3}
            name='ingredient3'
            onChange={this.handleChange}
            placeholder='enter an ingredient'
          />
          <Input 
            value={ingredient4}
            name='ingredient4'
            onChange={this.handleChange}
            placeholder='enter an ingredient'
          />
          <Input 
            value={ingredient5}
            name='ingredient5'
            onChange={this.handleChange}
            placeholder='enter an ingredient'
          />
          <div className='submit-btn'>
            <Button type='submit' variant="contained" color="primary">Find Recipes</Button>
          </div>
        </form>
      </section>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  fetchRecipes: (ingredients) => dispatch(fetchRecipes(ingredients))
})

export default withRouter(connect (null, mapDispatchToProps)(IngredientsForm))