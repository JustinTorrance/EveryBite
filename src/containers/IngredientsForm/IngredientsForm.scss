.ingredients-form {
  display: flex;
  flex-direction: column;
}