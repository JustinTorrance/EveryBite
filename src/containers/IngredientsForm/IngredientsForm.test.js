import { IngredientsForm, mapDispatchToProps, mapsta } from './IngredientsForm'
jest.mock('../../thunks/fetchRecipes')

describe('IngredientsForm', () => {

  describe('mapDispatchToProps', () => {
    it('should call mockDispatch when using a function from mapDispatchToProps', () => {
      const mockDispatch = jest.fn()
      const fetchRecipes = jest.fn()
      const actionToDispatch = fetchRecipes([{name: 'curry'}])
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchRecipes([{name: 'curry'}])
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
  
})



