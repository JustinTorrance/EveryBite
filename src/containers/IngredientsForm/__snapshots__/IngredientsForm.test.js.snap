// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientsForm should match the snapshot 1`] = `
"<section className=\\"ingredient-input-section\\">
  <WithStyles(Typography) variant=\\"title\\" color=\\"inherit\\">
    <div className=\\"instructions\\">
      <p>
        It&#39;s easy to begin!
      </p>
      <p>
        Simply enter the ingredients you have on hand,
      </p>
      <p>
        We&#39;ll find recipes that include all of your ingredients.
      </p>
    </div>
  </WithStyles(Typography)>
  <form className=\\"ingredients-form\\" onSubmit={[Function: onSubmit]}>
    <WithStyles(Input) value=\\"\\" name=\\"ingredient1\\" onChange={[Function]} placeholder=\\"enter an ingredient\\" />
    <WithStyles(Input) value=\\"\\" name=\\"ingredient2\\" onChange={[Function]} placeholder=\\"enter an ingredient\\" />
    <WithStyles(Input) value=\\"\\" name=\\"ingredient3\\" onChange={[Function]} placeholder=\\"enter an ingredient\\" />
    <WithStyles(Input) value=\\"\\" name=\\"ingredient4\\" onChange={[Function]} placeholder=\\"enter an ingredient\\" />
    <WithStyles(Input) value=\\"\\" name=\\"ingredient5\\" onChange={[Function]} placeholder=\\"enter an ingredient\\" />
    <div className=\\"submit-btn\\">
      <WithStyles(Button) type=\\"submit\\" variant=\\"contained\\" color=\\"primary\\">
        Find Recipes
      </WithStyles(Button)>
    </div>
  </form>
</section>"
`;
